staging/src/k8s.io/kubelet/pkg/apis/credentialprovider/types.go

```diff
@@ -32,6 +32,17 @@ type CredentialProviderRequest struct {
 	// credential provider plugin request. Plugins may optionally parse the image
 	// to extract any information required to fetch credentials.
 	Image string
+
+	// serviceAccountToken is the service account token bound to the pod for which
+	// the image is being pulled. This token is only sent to the plugin if the
+	// tokenAttributes.serviceAccountTokenAudience field is configured in the kubelet's credential provider configuration.
+	ServiceAccountToken string
```

Comment:
should we add  +featureGate=KubeletServiceAccountTokenForCredentialProviders?

Reply:
I tried adding this to the `CredentialProvider` new field (`tokenAttributes`) but I didn't see any difference in the generated OpenAPI doc. Let me know if you want me to add it!